### PR TITLE
fix: simplify security section

### PR DESCRIPTION
**Pull request by okdistribute**
- [x] Split out shared key section so it is easier to find
- [x] Update certificate-security section to use app id instead of urls
- [x] Cloud or Big Peer? I'm noticing these used interchangeably throughout the documentation, especially in this section. We should pick one and stick with it.
- [x] Snippets for shared key section should be generated from tests

### CODE REVIEW

File: docs/security/certificate-security.mdx [hunk 1]
```diff
@@ -3,15 +3,23 @@ title: 'Certificate Based Security'
...
 
-
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-> With the cloud platform, most use cases will not need to learn about this section. Instead, this is an advanced discussion of Ditto's underlying certificate, identity, and encryption implementation. However, if you are deploying an enterprise on-premise deployment of Ditto, you may be required to reference the following material.
+:::info
+This section contains an advanced discussion of Ditto's underlying certificate, identity, and
+encryption implementation. Most readers can skip this section.  However, if you are deploying an enterprise
+on-premise deployment of Ditto, you may be required to reference the following
```
Comment (thombles) [inline docs/security/certificate-security.mdx:12]: Existing text I know, but while we're here:

```suggestion
on-premises deployment of Ditto, you may be required to reference the following
```

File: docs/security/certificate-security.mdx [hunk 2]
```diff
@@ -107,133 +111,41 @@ An identity is a bundle of device and app-specific information:
...
-Ditto identities and public keys are distributed in the standard X.509
-certificate format. They do not directly contain potentially sensitive data such
-as access rules.
+## Certificate
 
-Ditto does not commit you to any workflow for managing identity data and
-certificates. When you are ready to start using production identities, contact
-the Ditto team through the Ditto Portal and we will help you set up the right CA
+Ditto identities and public keys are distributed in the standard X.509
+certificate format. By default, these are distributed through the Ditto Portal.
```
Comment (thombles) [inline docs/security/certificate-security.mdx:117]: "Ditto Cloud" or "Big Peer" would be more accurate than "Portal". It sounds a little like you could go onto the web interface and download an identity for a given device, similar to Apple's Developer portal, which you can't.

File: docs/security/certificate-security.mdx [hunk 3]
```diff
@@ -107,133 +111,41 @@ An identity is a bundle of device and app-specific information:
...
+rules dynamically, since they are not encoded in signed certificates. Another
+advantage is that all devices in the mesh can participate in syncing the data,
+which may help it propagate faster. If certain data is only accessible to a few
+privileged devices which are not often in range of each other, it will take
+longer for them to sync.
+
+The disadvantage is that an unprivileged user does have a device containing
+privileged data. A technically savvy user or phone thief may be able to gain
+access to not only their regular data, but also the more privileged data that
+they were never intended to able to view.
```
Comment (thombles) [inline docs/security/certificate-security.mdx:147]: Pre-existing typo

```suggestion
they were never intended to be able to view.
```

File: docs/security/identities.md [hunk 4]
```diff
@@ -6,48 +6,44 @@ sidebar_position: 2
...
+default settings and offers no real security. Development certificates should
+absolutely not be deployed to any production environments. Once you are ready to
+deploy, this can be swapped for a production identity where security is
+enforced.
+
+The Ditto Identity is one of the main inputs to the Ditto Constructor and
+controls how your local Ditto instance will communicate with the Ditto Cloud and
+other Ditto Peers.  Each Ditto Identity is associated to a single Ditto App via
+a globally unique AppId and may not be used across multiple Ditto apps. Each
+user that should be uniquely identified should have a distinct Ditto Identity. A
```
Comment (thombles) [inline docs/security/identities.md:21]: We could probably take out the "Each user that should be uniquely identified should have a distinct Ditto Identity." - it's kind of covered in the first para, and the word "user" seeds confusion about whether identity is a per-device or per-user-account concept (it's the former)

File: docs/security/identities.md [hunk 5]
```diff
@@ -6,48 +6,44 @@ sidebar_position: 2
...
-
-Generally the Ditto Cloud is not the actual Auth Provider, but a "proxy" to your own "source of truth" for identities, credential validation, and permissions.
-
-Depending on the Identity used, the Ditto instance may then request the Ditto Cloud (Big Peer) sign an x509 certificate, using the previously validated credentials and Authentication Token as proof of its identity. The Ditto Cloud will then return the signed x509 certificate. This certificate may then be presented to other Ditto peers to mutually establish trust and create encrypted communications channels.
```
Comment (thombles) [inline docs/security/identities.md:36]: I don't mind cutting back these details, but I reckon the new version should at least mention that by the end of this login process you can also sync P2P, in addition to the Ditto Cloud.

File: docs/security/identities.md [hunk 6]
```diff
@@ -6,48 +6,44 @@ sidebar_position: 2
...
+* `EnableCloudSync` - Whether Ditto should attempt to sync with `wss://<app-id>.cloud.ditto.live`. Only used by `OnlineWithAuthentication` and `OnlinePlayground` Identities. If you're hosting your own Ditto Cloud, this should be set to `false` and a custom SyncUrl provided to the Ditto `TransportConfig`
 * `WorkingDir` (DittoRoot) - A location where Ditto can cache Auth Tokens, x509 certificates, and other auth-related data. Typically the same as the working directory containing Ditto's local data store.
 * `AuthEventHandler` (Callback) - A type defined by the App developer which implements the `DittoAuthEventHandler` (`DittoAuthCallback`) Trait/Interface/Protocol. Typically responsible for obtaining and storing credentials like a username, password, or other token from the App's user and then supplying them to Ditto when authentication is required.
 
+For more detailed information on what is contained within Ditto Identity certificiates, see [Certificate Security](/security/certificate-security).
```
Comment (thombles) [inline docs/security/identities.md:40]:
```suggestion
For more detailed information on what is contained within Ditto Identity certificates, see [Certificate Security](/security/certificate-security).
```

File: docs/security/identities.md [hunk 7]
```diff
@@ -6,48 +6,44 @@ sidebar_position: 2
...
+* [OnlineWithAuthentication](/security/online-with-authentication/overview) - Use this identity for most production Apps. The `OnlineWithAuthentication` identity supports both Cloud and peer-to-peer sync with secure authentication, encryption, and access control.  The Ditto instance requests the Ditto Cloud sign an x509 certificate, using the previously validated credentials and Authentication Token as proof of its identity. The Ditto Cloud will then return the signed x509 certificate. This certificate may then be presented to other Ditto peers to mutually establish trust and create encrypted communications channels. Use typically requires configuring an Authentication Provider for your App within the Ditto Cloud portal, which can validate user-provided credentials. This identity also requires a developer-provided implementation of the `DittoAuthEventHandler` (`DittoAuthCallback`) interface. For a more thorough walkthrough for setting up authentication with your application, see the [tutorial](/tutorials/authentication/intro).
```
Comment (thombles) [inline docs/security/identities.md:45]: Ah I see this is where the x.509 details ended up. In the context of "here's a tour of the identity subtypes" this feels like a sudden dive in the deep end. I'd probably take out the middle bit from here, and consider a new heading on the Certificates page explaining where the certificates come from when you're using Online identity types.

File: docs/security/shared-key.mdx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+To generate a shared secret, use this command with OpenSSL:
+
+```console
+openssl genpkey -algorithm EC -pkeyopt ec_paramgen_curve:P-256 -outform pem | openssl pkcs8 -topk8 -nocrypt -outform der | base64 -w 0
+```
+
+At time of writing, this command works on Linux machines with an up-to-date
+OpenSSL but not with the LibreSSL distributed on Macs. See openssl version.
+
+<SnippetGroup name='shared-key' />
```
Comment (thombles) [inline docs/security/shared-key.mdx:27]: Were the snippets supposed to be copied across from the old page?

**Review by thombles — APPROVED** [review]
Thanks for the updates! Re: Ditto Cloud/Big Peer, personally I have a slight preference for "Ditto Cloud" when we're talking about something that only exists on our public cloud (e.g. playground apps, or related to the Portal) and the Big Peer for something that could also exist in a on-prem installation - e.g. webhooks or sync. No strong opinions though, they're pretty similar concepts.

File: docs/security/identities.md (hunk 4 above)
Comment (okdistribute) [reply docs/security/identities.md:21]: Thanks good point

**Comment by okdistribute** [comment]
Thanks @thombles for the review. I went with Big Peer to refer to what used to be called Hydra

**Pull request merged**